sympy/physics/control/lti.py

```diff
@@ -748,20 +838,32 @@ class Series(Basic):
 
     """
     def __new__(cls, *args, evaluate=False):
-        if not all(isinstance(arg, (TransferFunction, Parallel, Series)) for arg in args):
-            raise TypeError("Unsupported type of argument(s) for Series.")
-
-        obj = super().__new__(cls, *args)
-        obj._var = None
-        for arg in args:
-            if obj._var is None:
-                obj._var = arg.var
-            elif obj._var != arg.var:
-                raise ValueError("All transfer functions should use the same complex"
-                    " variable of the Laplace transform.")
-        if evaluate:
-            return obj.doit()
-        return obj
+        if len(args) == 0:
+            raise ValueError("Atleast 1 argument must be passed.")
+
+        try:
+            type_set = {arg.is_SISO for arg in args}
+            if len(type_set) != 1:
+                raise TypeError("Invalid system. `args` passed contain SISO as well "
+                    "as MIMO elements.")
+            var_set = {arg.var for arg in args}
+            if len(var_set) != 1:
+                raise ValueError("All transfer functions should use the same complex variable"
+                    f" of the Laplace transform. {len(var_set)} different values found.")
+        except AttributeError:
+            raise TypeError("Unsupported type of argument passed.")
```

Comment:
AttributeError should not be caught. Why is it necessary to catch this?

Whatever raises the error can be handled in a better way.